src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -412,6 +412,10 @@ internal int SerializeLogRecord(LogRecord logRecord)
                 cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, "eventId");
                 cursor = MessagePackSerializer.SerializeInt32(buffer, cursor, eventId.Id);
                 cntFields += 1;
+
+                cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, "eventName");
+                cursor = MessagePackSerializer.SerializeUnicodeString(buffer, cursor, eventId.Name);
+                cntFields += 1;
```

Comment:
Should we wrap this in a empty/null check like...

```csharp
                if (!string.IsNullOrEmpty(eventId.Name)
                {
                   cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, "eventName");
                   cursor = MessagePackSerializer.SerializeUnicodeString(buffer, cursor, eventId.Name);
                   cntFields += 1;
                }
```

Because name is optional: https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.eventid.-ctor?view=dotnet-plat-ext-6.0#microsoft-extensions-logging-eventid-ctor(system-int32-system-string)

Reply:
agree